crates/rustrace/src/main.rs

```diff
@@ -0,0 +1,69 @@
+use anyhow::{anyhow, bail, Result};
+use clap::Parser;
+use fork::{fork, Fork};
+use nix::{
+    errno::Errno,
+    sys::{
+        ptrace::{self, traceme, Options},
+        wait::waitpid,
+    },
+    unistd::Pid,
+};
+use syscalls::Sysno;
+
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    /// Command to trace
+    cmd: String,
+}
+
+fn main() -> Result<()> {
+    let args = Args::parse();
+
+    match fork() {
+        Ok(Fork::Parent(child)) => trace_child(child),
+        Ok(Fork::Child) => run_command_to_trace(&args.cmd),
+        Err(e) => Err(anyhow!("{e}")),
+    }
+}
+
+fn trace_child(child: i32) -> Result<()> {
+    let child = Pid::from_raw(child);
+    waitpid(Some(child), None)?;
+    // Jail the child
+    ptrace::setoptions(child, Options::PTRACE_O_EXITKILL)?;
```

Comment:
It wasn't apparent on first sight what does this mean. Could we maybe:

```suggestion
    // Kill the child in case the tracer exits
    ptrace::setoptions(child, Options::PTRACE_O_EXITKILL)?;
```

Though, we probably would remove this altogether if we'll be attaching to an externally passed pids.

Reply:
Good point! Sorry, I have been very terse with comments since I was mostly copying blindly :)

And yeah, as you say, this won't be necessary when attaching externally.